File: add_sample_data.js
const admin = require('firebase-admin');

// Initialize Firebase Admin SDK
const serviceAccount = {
  // This will use the default Firebase credentials from your project
};

if (!admin.apps.length) {
  try {
    admin.initializeApp({
      projectId: 'cyclesync-tracker'
    });
    console.log('Firebase Admin initialized successfully');
  } catch (error) {
    console.error('Error initializing Firebase Admin:', error);
    process.exit(1);
  }
}

const db = admin.firestore();
const userId = '4GEkV6hItgUgrZNPLMl3ES4mXOm2';

async function addSampleData() {
  try {
    console.log('Adding sample cycles...');
    
    // Add sample completed cycles
    const cycles = [
      {
        startDate: admin.firestore.Timestamp.fromDate(new Date('2024-06-15')),
        endDate: admin.firestore.Timestamp.fromDate(new Date('2024-06-19')),
        length: 4,
        flowIntensity: 'medium',
        symptoms: ['cramps', 'mood_swings'],
        notes: 'Normal cycle, moderate flow',
        isComplete: true,
        createdAt: admin.firestore.Timestamp.fromDate(new Date('2024-06-15')),
        updatedAt: admin.firestore.Timestamp.fromDate(new Date('2024-06-19'))
      },
      {
        startDate: admin.firestore.Timestamp.fromDate(new Date('2024-07-12')),
        endDate: admin.firestore.Timestamp.fromDate(new Date('2024-07-17')),
        length: 5,
        flowIntensity: 'heavy',
        symptoms: ['cramps', 'fatigue', 'bloating'],
        notes: 'Heavy flow cycle with stronger symptoms',
        isComplete: true,
        createdAt: admin.firestore.Timestamp.fromDate(new Date('2024-07-12')),
        updatedAt: admin.firestore.Timestamp.fromDate(new Date('2024-07-17'))
      },
      {
        startDate: admin.firestore.Timestamp.fromDate(new Date('2024-08-09')),
        endDate: admin.firestore.Timestamp.fromDate(new Date('2024-08-13')),
        length: 4,
        flowIntensity: 'light',
        symptoms: ['mood_swings'],
        notes: 'Light cycle with minimal symptoms',
        isComplete: true,
        createdAt: admin.firestore.Timestamp.fromDate(new Date('2024-08-09')),
        updatedAt: admin.firestore.Timestamp.fromDate(new Date('2024-08-13'))
      },
      {
        startDate: admin.firestore.Timestamp.fromDate(new Date('2024-09-05')),
        endDate: admin.firestore.Timestamp.fromDate(new Date('2024-09-09')),
        length: 4,
        flowIntensity: 'medium',
        symptoms: ['cramps', 'headache'],
        notes: 'Regular cycle with headaches',
        isComplete: true,
        createdAt: admin.firestore.Timestamp.fromDate(new Date('2024-09-05')),
        updatedAt: admin.firestore.Timestamp.fromDate(new Date('2024-09-09'))
      },
      {
        startDate: admin.firestore.Timestamp.fromDate(new Date('2024-10-02')),
        endDate: admin.firestore.Timestamp.fromDate(new Date('2024-10-06')),
        length: 4,
        flowIntensity: 'medium',
        symptoms: ['cramps'],
        notes: 'Normal cycle',
        isComplete: true,
        createdAt: admin.firestore.Timestamp.fromDate(new Date('2024-10-02')),
        updatedAt: admin.firestore.Timestamp.fromDate(new Date('2024-10-06'))
      },
      {
        startDate: admin.firestore.Timestamp.fromDate(new Date('2024-10-29')),
        endDate: admin.firestore.Timestamp.fromDate(new Date('2024-11-02')),
        length: 4,
        flowIntensity: 'medium',
        symptoms: ['cramps', 'mood_swings'],
        notes: 'Recent cycle',
        isComplete: true,
        createdAt: admin.firestore.Timestamp.fromDate(new Date('2024-10-29')),
        updatedAt: admin.firestore.Timestamp.fromDate(new Date('2024-11-02'))
      }
    ];

    // Add cycles to Firestore
    for (let i = 0; i < cycles.length; i++) {
      const cycleRef = db.collection('users').doc(userId).collection('cycles').doc();
      await cycleRef.set(cycles[i]);
      console.log(`Added cycle ${i + 1}: ${cycles[i].startDate.toDate().toDateString()} - ${cycles[i].endDate.toDate().toDateString()}`);
    }

    console.log('Adding sample daily logs...');

    // Add sample daily logs
    const dailyLogs = [
      {
        date: admin.firestore.Timestamp.fromDate(new Date('2024-11-15')),
        mood: 'happy',
        energyLevel: 8,
        sleepQuality: 7,
        symptoms: ['cramps'],
        notes: 'Feeling good today',
        weight: 65.5,
        waterIntake: 2000,
        exercise: 'walking',
        createdAt: admin.firestore.Timestamp.now(),
        updatedAt: admin.firestore.Timestamp.now()
      },
      {
        date: admin.firestore.Timestamp.fromDate(new Date('2024-11-16')),
        mood: 'neutral',
        energyLevel: 6,
        sleepQuality: 6,
        symptoms: ['bloating', 'mood_swings'],
        notes: 'Average day',
        weight: 65.8,
        waterIntake: 1800,
        exercise: 'yoga',
        createdAt: admin.firestore.Timestamp.now(),
        updatedAt: admin.firestore.Timestamp.now()
      },
      {
        date: admin.firestore.Timestamp.fromDate(new Date('2024-11-17')),
        mood: 'happy',
        energyLevel: 9,
        sleepQuality: 8,
        symptoms: [],
        notes: 'Great energy today!',
        weight: 65.2,
        waterIntake: 2200,
        exercise: 'running',
        createdAt: admin.firestore.Timestamp.now(),
        updatedAt: admin.firestore.Timestamp.now()
      }
    ];

    // Add daily logs to Firestore
    for (let i = 0; i < dailyLogs.length; i++) {
      const logRef = db.collection('users').doc(userId).collection('daily_logs').doc();
      await logRef.set(dailyLogs[i]);
      console.log(`Added daily log ${i + 1}: ${dailyLogs[i].date.toDate().toDateString()}`);
    }

    console.log('✅ Sample data added successfully!');
    console.log(`Added ${cycles.length} cycles and ${dailyLogs.length} daily logs for user ${userId}`);

  } catch (error) {
    console.error('Error adding sample data:', error);
  } finally {
    process.exit(0);
  }
}

addSampleData();

File: add_sample_data.dart
import 'dart:io';
import 'package:firebase_core/firebase_core.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'lib/firebase_options.dart';

void main() async {
  // Initialize Firebase
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );

  final firestore = FirebaseFirestore.instance;
  const userId = '4GEkV6hItgUgrZNPLMl3ES4mXOm2'; // The user ID from logs

  // Adding sample cycle data for user: $userId

  // Sample cycle data for the last 6 months
  final sampleCycles = [
    {
      'id': 'cycle_1',
      'start_date': DateTime(2025, 2, 1),
      'end_date': DateTime(2025, 2, 28),
      'flow': 'medium',
      'mood_level': 4,
      'energy_level': 3,
      'pain_level': 2,
      'symptoms': ['cramps', 'headache'],
      'notes': 'First cycle tracked',
      'created_at': DateTime(2025, 2, 1),
    },
    {
      'id': 'cycle_2',
      'start_date': DateTime(2025, 3, 2),
      'end_date': DateTime(2025, 3, 30),
      'flow': 'heavy',
      'mood_level': 3,
      'energy_level': 2,
      'pain_level': 4,
      'symptoms': ['cramps', 'fatigue', 'mood_swings'],
      'notes': 'Heavy flow cycle',
      'created_at': DateTime(2025, 3, 2),
    },
    {
      'id': 'cycle_3',
      'start_date': DateTime(2025, 4, 1),
      'end_date': DateTime(2025, 4, 27),
      'flow': 'light',
      'mood_level': 5,
      'energy_level': 4,
      'pain_level': 1,
      'symptoms': ['bloating'],
      'notes': 'Light and easy cycle',
      'created_at': DateTime(2025, 4, 1),
    },
    {
      'id': 'cycle_4',
      'start_date': DateTime(2025, 5, 3),
      'end_date': DateTime(2025, 5, 31),
      'flow': 'medium',
      'mood_level': 4,
      'energy_level': 4,
      'pain_level': 2,
      'symptoms': ['cramps', 'headache', 'breast_tenderness'],
      'notes': 'Regular cycle',
      'created_at': DateTime(2025, 5, 3),
    },
    {
      'id': 'cycle_5',
      'start_date': DateTime(2025, 6, 2),
      'end_date': DateTime(2025, 6, 29),
      'flow': 'medium',
      'mood_level': 3,
      'energy_level': 3,
      'pain_level': 3,
      'symptoms': ['cramps', 'fatigue', 'irritability'],
      'notes': 'Average cycle',
      'created_at': DateTime(2025, 6, 2),
    },
    {
      'id': 'cycle_6',
      'start_date': DateTime(2025, 7, 1),
      'end_date': DateTime(2025, 7, 28),
      'flow': 'light',
      'mood_level': 4,
      'energy_level': 5,
      'pain_level': 1,
      'symptoms': ['mood_swings'],
      'notes': 'Great energy this cycle',
      'created_at': DateTime(2025, 7, 1),
    },
  ];

  try {
    // Add each cycle to Firestore
    for (int i = 0; i < sampleCycles.length; i++) {
      final cycleData = sampleCycles[i];
      
      await firestore
          .collection('users')
          .doc(userId)
          .collection('cycles')
          .doc(cycleData['id'] as String)
          .set({
        'start': Timestamp.fromDate(cycleData['start_date'] as DateTime),
        'end': Timestamp.fromDate(cycleData['end_date'] as DateTime),
        'flow': cycleData['flow'],
        'mood_level': cycleData['mood_level'],
        'energy_level': cycleData['energy_level'],
        'pain_level': cycleData['pain_level'],
        'symptoms': cycleData['symptoms'],
        'notes': cycleData['notes'],
        'created_at': Timestamp.fromDate(cycleData['created_at'] as DateTime),
      });
      
      // Added cycle ${i + 1}/6: ${cycleData['id']}
    }

    // Add some daily logs too
    final sampleDailyLogs = [
      {
        'date': DateTime(2025, 8, 5),
        'mood': 4.0,
        'energy': 3.5,
        'pain': 1.0,
        'symptoms': ['headache'],
        'notes': 'Slight headache today',
      },
      {
        'date': DateTime(2025, 8, 7),
        'mood': 5.0,
        'energy': 4.0,
        'pain': 0.0,
        'symptoms': [],
        'notes': 'Great day!',
      },
      {
        'date': DateTime(2025, 8, 9),
        'mood': 3.0,
        'energy': 2.5,
        'pain': 2.0,
        'symptoms': ['fatigue', 'cramps'],
        'notes': 'Feeling tired',
      },
    ];

    for (int i = 0; i < sampleDailyLogs.length; i++) {
      final logData = sampleDailyLogs[i];
      final dateStr = (logData['date'] as DateTime).toIso8601String().split('T')[0];
      
      await firestore
          .collection('users')
          .doc(userId)
          .collection('daily_logs')
          .doc(dateStr)
          .set({
        'date': Timestamp.fromDate(logData['date'] as DateTime),
        'mood': logData['mood'],
        'energy': logData['energy'],
        'pain': logData['pain'],
        'symptoms': logData['symptoms'],
        'notes': logData['notes'],
        'created_at': Timestamp.now(),
      });
      
      // Added daily log ${i + 1}/3: $dateStr
    }

    // Successfully added sample data!
    // Analytics should now show:
    // • 6 completed cycles
    // • Cycle length trends
    // • Symptom patterns
    // • Predictions with confidence
    // • Wellbeing trends
    // Restart the app to see the analytics!

  } catch (e) {
    // Error adding sample data: $e
  }

  exit(0);
}
